src/extract.jl

```diff
@@ -285,7 +285,15 @@ function git_object_hash(
     kind::AbstractString,
     HashType::DataType,
```

Comment:
I think you can just change the signature to `::Type{HashType} …. where {HashType}` instead, no?

Reply:
That might be better then, yeah.
I could also just instantiate `HashType` to a `context` at the outset?
I thought maybe the lack of specialization was intentional.

Reply:
The issue with the latter approach is that you need many context instances while computing git-style tree hashes, not just one context object.

Reply:
Makes sense, thanks.

I changed to `::Type{HashType}, ...) where HashType` now. I also changed the branching for calling `git_tree_hash` to hopefully eliminate any other instabilities at their origin.